src/ape/api/providers.py

```diff
@@ -133,6 +134,27 @@ def decode(cls, data: dict) -> "ReceiptAPI":
         ...
 
 
+@abstractdataclass
+class BlockAPI:
+    base_fee_per_gas: Optional[int] = None
+    difficulty: int
```

Comment:
Same idea, but with `consensus_data: dict` to abstract out difficulty/PoA stuff, miner/coinbase, etc.